### PR TITLE
Improve cart functionality

### PR DESCRIPTION
**Pull request by lazreg87**
Refactor the `add product` and `remove product` methods to now increment, decrement existing product lines. A product line is deemed as identical if the productId and subproducts (regardless of order) are the same. Otherwise, we produce a new product line in the cart

### CODE REVIEW

File: src/functionality/cart.js [hunk 1]
```diff
@@ -12,19 +12,28 @@ class Cart {
     return this._cart.findIndex(cartItem => cartItem.productId === productId);
   }
 
+  // Cart is valid if all item contain at least productId and quantity
   _isCartValid() {
+    let valid = true;
     this._cart.forEach((item) => {
```
Comment (NicolasBonduel) [inline src/functionality/cart.js]: This loop shouldn't be a `forEach`, as forEach is not breakable.

I did not test it, but I think you can use something like:
```javascript
return !this._cart.some((item) => {
    return (!item.productId || !item.quantity);
});
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

File: src/functionality/cart.js [hunk 2]
```diff
@@ -12,19 +12,28 @@ class Cart {
...
+    return valid;
   }
 
   _saveCart() {
     store.set('cart', this._cart);
   }
 
+  // Check to see if two subproduct lists contain the same items. Doesn't care about order
+  // eslint-disable-next-line class-methods-use-this
+  _compare(listA = [], listB = []) {
```
Comment (NicolasBonduel) [inline src/functionality/cart.js]: Can this be a static method?

Comment (lazreg87) [reply src/functionality/cart.js]: Yes

File: src/functionality/cart.js (hunk 1 above)
Comment (lazreg87) [reply src/functionality/cart.js]: I agree

**Pull request merged**